room/compiler-processing/src/main/java/androidx/room/compiler/processing/XProcessingStep.kt

```diff
@@ -66,18 +67,22 @@ interface XProcessingStep {
         )
     }
 
-    fun executeInKsp(env: XProcessingEnv) {
+    fun executeInKsp(env: XProcessingEnv): List<KSAnnotated> {
         check(env is KspProcessingEnv)
+        val elementMap = mutableMapOf<XTypeElement, KSAnnotated>()
         val args = annotations().associateWith { annotation ->
             val elements = env.resolver.getSymbolsWithAnnotation(
                 annotation.java.canonicalName
             ).filterIsInstance<KSClassDeclaration>()
                 .map {
-                    env.requireTypeElement(it.qualifiedName!!.asString())
+                    val element = env.requireTypeElement(it.qualifiedName!!.asString())
+                    elementMap[element] = it
+                    element
```

Comment:
you could also cast this to KspTypeElement to get the declaration out of it but this soolution is fine too.

Reply:
I think it's more cohesive to do that, indeed